rpc/client.go

```diff
@@ -0,0 +1,42 @@
+package rpc
```

Comment:
gw pikir `proto`, `rpc` and `util` package ini gak perlu jadi package yg berbeda. it can be part of `raft` package.

kalau mau tetap dipisah, better to do something like `rpc.NewClient()` instead of `rpc.NewRPCClient()`. Ketika `NewClient` dipanggil dari package `rpc`, udah ada context tambahan bahwa itu adalah client rpc.

tapi kalau digabung, bisa lebih sederhana di dalam package raft.

```
type Raft struct {
  rpc RPCClient
}

func NewRaft() {
  r := &Raft{
    rpc: newRPCClient(), 
  }
}
```

`newRPCClient()` bisa private utk package `raft`. jadi bisa naikin module cohession, dan ngurangin coupling ke package lain. lagi pula yg akan pakai rpc client cuma module raft aja kan sebenernya.

Reply:
thanks feedbacknya. setuju. tapi untuk yang 'client', soalnya gue kepikirannya nanti ada client lain untuk usecase lain yang pakai ini (e.g. untuk call leader untuk nambahin log)